snap/info_snap_yaml_test.go

```diff
@@ -39,14 +41,801 @@ type: app
 `)
 
 func (s *InfoSnapYamlTestSuite) TestSimple(c *C) {
-	info, err := InfoFromSnapYaml(mockYaml)
+	info, err := snap.InfoFromSnapYaml(mockYaml)
 	c.Assert(err, IsNil)
 	c.Assert(info.Name, Equals, "foo")
 	c.Assert(info.Version, Equals, "1.0")
-	c.Assert(info.Type, Equals, TypeApp)
+	c.Assert(info.Type, Equals, snap.TypeApp)
 }
 
 func (s *InfoSnapYamlTestSuite) TestFail(c *C) {
-	_, err := InfoFromSnapYaml([]byte("random-crap"))
+	_, err := snap.InfoFromSnapYaml([]byte("random-crap"))
 	c.Assert(err, ErrorMatches, "(?m)info failed to parse:.*")
 }
+
+type YamlSuite struct{}
+
+var _ = Suite(&YamlSuite{})
+
+func (s *YamlSuite) TestUnmarshalGarbage(c *C) {
+	_, err := snap.InfoFromSnapYaml([]byte(`"`))
+	c.Assert(err, ErrorMatches, ".*: yaml: found unexpected end of stream")
+}
+
+func (s *YamlSuite) TestUnmarshalEmpty(c *C) {
+	info, err := snap.InfoFromSnapYaml([]byte(``))
+	c.Assert(err, IsNil)
+	c.Assert(info.Plugs, HasLen, 0)
+	c.Assert(info.Slots, HasLen, 0)
+	c.Assert(info.Apps, HasLen, 0)
+}
+
+// Tests focusing on plugs
+
+func (s *YamlSuite) TestUnmarshalStandaloneImplicitPlug(c *C) {
+	// NOTE: yaml content cannot use tabs, indent the section with spaces.
+	info, err := snap.InfoFromSnapYaml([]byte(`
+name: snap
+plugs:
+    network-client:
+`))
+	c.Assert(err, IsNil)
+	c.Check(info.Name, Equals, "snap")
+	c.Check(info.Plugs, HasLen, 1)
+	c.Check(info.Slots, HasLen, 0)
+	plug := info.Plugs["network-client"]
+
+	c.Assert(plug, Not(IsNil))
```

Comment:
I agree somewhat. There's a tricky balance between making each test easy to understand and standalone and making all tests terse and non-repetitive. I think we could use the ``DeepEquals`` approach for comparison, it would certainly be easier to parse visually than the swarm of separate tests. I would not like to block on this because we should get a chance to use this first and the tests were time consuming as they already are.